### DISCUSSION TITLE
walrus-tar

### DISCUSSION THREAD

**Discussion by nextuser** [creatorsand/.github#138] [Ideas]
# 一句话 IDEA：
- 将小文件打包成tar ，上传 到walrus
- 将小文件从walrus下载，从tar 提取文件

# 我的角色：
- 开发

# 其他角色需求（不包含发起人）：
- 无
# github
https://github.com/nextuser/walrus-tar

# 社群昵称：
科学减肥-jinglin

# 个人简介：
c/c++ 

# 产品是给谁用的：
普通用户需要存储图片 文本 pdf等小文件的时候，需要整体打包成一个blob

# 其他产品有什么痛点，用户为什么要选择你的产品？
1.目前walrus最小的文件都几十兆，小文件也会打padding，存储浪费。对于普通用户是不划算的。
节省文件存储开销。 目前walrus 一个小文件都要扩展成几十mb， 实际我们需要存储的文件往往都比较小
2. 实际很多内容是关联的，不希望独立存储，希望作为整体存储。 可以打成一个包整体存储。

**Comment by nextuser** [comment]
# 代码仓库
- [https://github.com/nextuser/deposit-bonus](https://github.com/nextuser/deposit-bonus)
# demo地址
- [https://deposit-bonus-gamma.vercel.app](https://deposit-bonus-gamma.vercel.app)

**Comment by v1xingyue** [comment]
ai打分

[a comment by an automated account is not shown]

**Comment by nextuser** [comment]
1. tar 包功能只是对walrus的缺陷的一个补充吧，我怀疑可能官方有余暇的时候，能自己做进去
技术实现比较简单，就是简单的读取文件写入tar 的数据流，本来还可以加入gzip 压缩，这次没有太多时间，我就没做了。
2. 做tar包功能 隐含了一个价值，我们读取了一次文件名 文件路径
如果做成在线上传工具，我们可能生成文件索引，实现线上的文件搜索功能。 对比：如果用户自己打包的blob，walrus是不会解析文件关键字  
3.  其实真正具有商用价值的，是做一个类似网盘一样的应用。
walrus要实现网盘功能，就需要解决小文件问题，tar包能解决。也需要文件重新封包之后，怎么把关联到原来blob的费用转移到新blob上。（比如old blob 的storage 拥有的epoch能否签约到new blob上来），目前的walrus 没开放相关接口吧。







---- Replied Message ----
| From | ***@***.***> |
| Date | 12/22/2024 12:51 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [CreatorsDAO/.github] walrus-tar (Discussion #138) |

根据项目描述，我对该项目进行评分，并提供评分理由。

评分：70/100
评分理由：
优点：

解决实际问题：项目明确指出了当前 Walrus 存储系统的小文件存储浪费问题，并提供了一种解决方案，即通过打包小文件成 tar 以减少存储的开销。这是一个现实的问题，并且项目提供了一个有效的解决方法。

目标用户清晰：项目针对普通用户存储小文件的需求，目标明确，用户群体定位清晰。

简单可行的技术实现：利用 tar 格式进行打包，对于文件整体性存储以及节省空间有实际效果。

缺点：

创新性有限：虽然解决了现有问题，但项目本质上是对现有工具和技术的简单结合，没有突破性的创新或技术上的亮点。

缺乏详细的技术说明：项目描述中缺乏关于如何在 Walrus 上进行打包和解包的具体技术细节。这可能会影响评委和潜在用户对项目技术实现的信任。

用户体验考虑不足：项目目前的方案对于终端用户来说，可能在可操作性和易用性上有不足之处，如何简化用户操作流程和提高用户体验未在描述中体现。

市场竞争和独特性没有明确定义：虽指出了其他产品的痛点，但未能详细描述为何用户一定会选择此方案，其他替代方案是什么。

协作开发及市场推广：项目中明确说明只有一个开发者，没有提及是否会在项目推广、用户获取等方面进行更多的协作与规划。

综上所述，这个项目具有解决实际问题的潜力并且目标用户明确，但在创新性、技术细节和用户体验上还有改进的空间。为了在评分上得到更高的分数，项目可以进一步扩展技术实现的细节，提升用户体验，增强产品独特性和市场竞争力。
